gqcp/include/Processing/Properties/expectation_values.hpp

```diff
@@ -89,6 +88,46 @@ std::array<double, Components> calculateExpectationValue(const SQTwoElectronOper
 }
 
 
+/**
+ *  Calculate the expectation value for spin squared
+ * 
+ *  @tparam Scalar      the scalar type
+ * 
+ *  @param one_rdms        all the one electron density matrices
+ *  @param two_rdms        all the two electron density matrices
+ *
+ *  @return expectation value of spin squared
+ */
+template <typename Scalar>
+double calculateSpinSquared(const OneRDMs<Scalar>& one_rdms, const TwoRDMs<Scalar>& two_rdms) {
+    double sz = calculateSpinZ(one_rdms);
+    double s_squared = -sz;
+    size_t K = one_rdms.one_rdm.dimension();
```

Comment:
const
```suggestion
    const size_t K = one_rdms.one_rdm.dimension();
```

Reply:
Also, could you provide the implementation of `OneRDMs.dimension()` and check the whole library for these occurrences?

Reply:
I do not understand the request.

Reply:
`OneRDM` has a method called `dimension()`, could you implement `dimension()` for `OneRDMs` too?